### PR TITLE
Llvm14config

### PR DESCRIPTION
**Pull request by nishant-sachdeva**

### CODE REVIEW

File: src/test-suite/sanity_check.sh.cmake [hunk 1]
```diff
@@ -25,117 +25,117 @@ else
...
-    mkdir -p ${TEMP}
-    mv *${VIR_FILE} ${TEMP}/
-
-
-    d=$(diff <(sed -e 's/^ *#[0-9]* *//g' ${ORIG_FILE}) <(sed -e 's/^ *#[0-9]* *//g' ${TEMP}/${VIR_FILE}))
-    if [ "$d" == "" ]
-    then
-        echo -e "${GREEN}${BOLD}[Test Passed] Vectors of  Oracle and Current version of p-level are Identical.${NC}"
+VOCAB_PATH="/home/intern23002/iitH/ir2vec/IR2Vec/vocabulary/seedEmbedding_1500E_300D_250batches5.0margin.txt"
+IR2VEC_PATH="/home/intern23002/iitH/ir2vec/IR2Vec/build/bin/ir2vec"
```
Comment (svkeerthy) [inline src/test-suite/sanity_check.sh.cmake:29]: Why is this needed? Please make it like the earlier version.

File: vocabulary/seedEmbedding_1500E_300D.txt
Comment (svkeerthy) [inline vocabulary/seedEmbedding_1500E_300D.txt:1]: One of these two is to be removed?

**Review by svkeerthy — CHANGES_REQUESTED** [review]

**Comment by svkeerthy** [comment]
You may need to fix `.github/workflows/test.yml` as well.

File: src/test-suite/generateOracle.sh [hunk 2]
```diff
@@ -4,25 +4,38 @@
...
 mkdir -p ${DEST_FOLDER_SYM_ONDEMAND}
 mkdir -p ${DEST_FOLDER_FA_ONDEMAND}
 
 # Update the BUILD to use
-LLVM_BUILD=
+LLVM_BUILD="/usr"
 
 #Update IR2Vec Path to use
-IR2VEC_PATH=
+IR2VEC_PATH="/home/intern23002/iitH/ir2vec/IR2Vec/build/bin/ir2vec"
```
Comment (svkeerthy) [inline src/test-suite/generateOracle.sh:29]: Remove the hard-coded paths

File: src/test-suite/generateOracle.sh [hunk 3]
```diff
@@ -44,13 +57,29 @@ functions=("main" "buildMatchingMachine" "search" "BellamFord" "BFS" "isBCUtil"
 	"selectKItems" "getMinDiceThrows" "countSort" "subset_sum" "SolveSudoku" "SCC" "solveKTUtil" "topologicalSort" "transitiveClosure" "insertSuffix" "tugOfWar" "isUgly" "Union" "printVertexCover"
 	"findMaxProfit" "solveWordWrap")
 
+# define a counter variable to count the number of files
+cat index-${SEED_VERSION}.files | wc -l
+echo "generating P level files"
+while IFS= read -r d; do
+	${IR2VEC_PATH} -sym -vocab=${VOCAB_PATH} -level p -o ${DEST_FOLDER_SYM_P}/ir2vec.txt ${d}
+	${IR2VEC_PATH} -fa -vocab=${VOCAB_PATH} -level p -o ${DEST_FOLDER_FA_P}/ir2vec.txt ${d}
```
Comment (svkeerthy) [inline src/test-suite/generateOracle.sh:65]: Please redirect outputs to /dev/null

File: src/test-suite/generate_llfiles.sh [hunk 4]
```diff
@@ -5,20 +5,20 @@
...
 
 # Update the BUILD to use
-LLVM_BUILD=
+LLVM_BUILD="/usr"
 
-if [ -z ${LLVM_BUILD} ]; then
-	echo "Enter the llvm build path.."
-	exit
-fi
+# if [ -z ${LLVM_BUILD} ]; then
```
Comment (svkeerthy) [inline src/test-suite/generate_llfiles.sh:15]: Pl revert back

File: src/test-suite/generate_llfiles.sh [hunk 5]
```diff
@@ -5,20 +5,20 @@
...
+# 	echo "Enter the llvm build path.."
+# 	exit
+# fi
 
-for d in ${SRC_WD}/*.c ${SRC_WD}/*.cpp ${SRC_WD}/*.cc; do
-	name=$(basename ${d}) && oname=${name%.*} && ${LLVM_BUILD}/bin/clang -S -emit-llvm -Xclang -disable-O0-optnone ${d} -o ${DEST_FOLDER}/${oname}.ll &
+for d in ${SRC_WD}/*.cpp; do
+	name=$(basename ${d}) && oname=${name%.*} && ${LLVM_BUILD}/bin/clang-14 -S -emit-llvm -Xclang -disable-O0-optnone ${d} -o ${DEST_FOLDER}/${oname}.ll &
```
Comment (svkeerthy) [inline src/test-suite/generate_llfiles.sh:21]: Use clang-14++

File: src/test-suite/sanity_check.sh.cmake [hunk 6]
```diff
@@ -25,117 +25,117 @@ else
...
+
+perform_vector_comparison() {
```
Comment (svkeerthy) [inline src/test-suite/sanity_check.sh.cmake:31]: Please merge the functions into one and reuse.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: src/test-suite/generate_llfiles.sh (hunk 4 above)
Comment (nishant-sachdeva) [reply src/test-suite/generate_llfiles.sh:15]: the LLVM_BUILD path is never used in the script. So, I've removed it. Please check if it needs to be reinstated nonetheless

File: README.md [hunk 7]
```diff
@@ -6,18 +6,15 @@ Please see [here](https://compilers.cse.iith.ac.in/projects/ir2vec/) for more de
...
 
-![LLVM](https://img.shields.io/badge/LLVM-v12.0.0-blue)
+![LLVM](https://img.shields.io/badge/LLVM-v14.0.0-blue)
 ![Tests](https://github.com/IITH-Compilers/IR2Vec/workflows/Tests/badge.svg)
 ![Publish](https://github.com/IITH-Compilers/IR2Vec/workflows/Publish/badge.svg)
 ![pre-commit checks](https://github.com/IITH-Compilers/IR2Vec/workflows/pre-commit%20checks/badge.svg)
 
 ![Image](images/ir2vec.jpg)
 
-## { create a table that mentions the llvm version to be used in each branch/release }
```
Comment (svkeerthy) [inline README.md]: Please add the table here.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

**Review by svkeerthy — APPROVED** [review]
